### ISSUE TITLE
Provide ITextFormatter Imlementation for Stdout

### ISSUE THREAD

**Issue by JarrodJ83** [DataDog/serilog-sinks-datadog-logs#37]
We currently have all of our containers set to write to stdout by using `WriteTo.Console` in Serilog. [Per the DataDog documentation ](https://docs.datadoghq.com/logs/log_collection/csharp/?tab=serilog) we should use the `JsonFormatter`. When using this in combination writing to stdout there are a couple issues:

- Logs do not show up correctly in the Log Explorer
![image](https://user-images.githubusercontent.com/248166/76800592-5f592380-67aa-11ea-95a6-f05f65fa39c5.png)
- Level is not interpreted correctly
![image](https://user-images.githubusercontent.com/248166/76800930-ffaf4800-67aa-11ea-9b34-0c9ebbb07a09.png)
- Correlation between logs and traces does not work (No "See trace in APM" link)
![image](https://user-images.githubusercontent.com/248166/76800839-dabad500-67aa-11ea-90fc-9aa1dfb7469e.png)

We were previously using ES/Kibana for logging so we were writing to the console using `WriteTo.Console(new ElasticsearchJsonFormatter())`. If I switch to this everything works! 

My fear is that eventually this formatter may change in a way that is not compatible with DataDog. It would be great if you could provide a DataDog implementation of `Serilog.Formatting.ITextFormatter` so that we are guaranteed this functionality will not break in the future.

**Comment by ogaca-dd** [comment]
@JarrodJ83 

Thank you for reporting this issue.
`JsonFormatter` uses the field `RenderedMessage` instead of `message` and having a custom 
 `Serilog.Formatting.ITextFormatter` would solve the issue. 
I have created a card in our backlog. I will update this issue when I will have more information.

**Comment by JarrodJ83** [comment]
Sounds great, @ogaca-dd! Thanks for the udpate.

**Comment by pergardebrink** [comment]
Yes! I would also really like to have a Datadog provided ITextFormatter for JSON! 

We needed to get structured logging from our AWS lambdas from stdout so that CloudWatch shipping would work, so we didn't have to use the sink (that just adds network latency), and I ended up with writing a custom Datadog ITextFormatter instead.

I found some guidance at Serilog where they pointed here: https://github.com/serilog/serilog-formatting-compact/blob/dev/src/Serilog.Formatting.Compact/Formatting/Compact/RenderedCompactJsonFormatter.cs and looked a bit at what JSON Datadog seems to expect in the LogFormatter in this repository.

So the following gist is a part of what I hacked together in an internal NuGet package we have for use in our Serilog+Lambda+Datadog, until something officially supported by Datadog would be released:
https://gist.github.com/pergardebrink/66d31e4d1b62e3d36c38ca999b75bfc7
